src/App.tsx

```diff
@@ -1,26 +1,128 @@
+import { FilterTypes } from './types/FilterTypes';
+import classNames from 'classnames';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
 
-const USER_ID = 0;
+import { filterTodos } from './helper/utilsFunctions';
+import { ErrorMessage } from './types/ErrorMessage';
+import { ErrorNotifaction } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodos, setSelectedTodos] = useState<FilterTypes>(
+    FilterTypes.All,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeletedTodoHasLoader, setIsDeletedTodoHasLoader] = useState(false);
+  const [isTodoRenaming, setIsTodoRenaming] = useState(false);
+  const [renameTodoTitle, setRenameTodoTitle] = useState('empty');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const areTodosExist = !!todos.length;
+  const completedIds = todos
+    .filter(todo => todo.completed)
+    .map(todo => todo.id);
+
+  const filteredTodos = filterTodos(todos, selectedTodos);
+
+  const notCompletedTodosCount = todos.filter(todo => !todo.completed).length;
+  const isAnyCompletedTodos = notCompletedTodosCount === filteredTodos.length;
+  const areAllTodosCompleted = notCompletedTodosCount === 0;
+  const allNotCompletedTodos = todos.filter(todo => todo.completed === false);
+
+  const handleError = (error: string) => {
+    setErrorMessage(error);
+
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handleError(ErrorMessage.LoadingError);
+      });
+  }, []);
+
+  function handleDeleteTodoCLick(todoId: number) {
+    setIsDeletedTodoHasLoader(true);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(prevTodos =>
+          prevTodos.filter(currTodo => todoId !== currTodo.id),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.DeleteTodoError);
+      })
+      .finally(() => setIsDeletedTodoHasLoader(false));
+  }
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div
+      className={classNames('todoapp', {
+        'has-error': Boolean(errorMessage.length),
+      })}
+    >
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          areAllTodosCompleted={areAllTodosCompleted}
+          setTodos={setTodos}
```

Comment:
Consider creating handler function because pass useState setter is a bad practice